src/scheduler.rs

```diff
@@ -323,6 +323,33 @@ pub(crate) struct Scheduler {
     recently_seen_loop: RecentlySeenLoop,
 }
 
+async fn download_msgs(context: &Context, imap: &mut Imap) -> Result<()> {
+    let msg_ids = context
+        .sql
+        .query_map(
+            "SELECT msg_id FROM download",
+            (),
+            |row| {
+                let msg_id: MsgId = row.get(0)?;
+                Ok(msg_id)
+            },
+            |rowids| {
+                rowids
+                    .collect::<std::result::Result<Vec<_>, _>>()
+                    .map_err(Into::into)
+            },
+        )
+        .await?;
+
+    for msg_id in msg_ids {
+        download_msg(context, msg_id, imap)
+            .await
+            .context("failed to download message {msg_id}")?;
```

Comment:
Should we skip the remaining messages? Maybe remember the error and return it in the end?